logging/resource.go

```diff
@@ -0,0 +1,266 @@
+	_, instance := os.LookupEnv("GAE_INSTANCE")
+
+	return service && version && instance
+}
+
+func detectAppEngineResource() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	if projectID == "" {
+		projectID = os.Getenv("GOOGLE_CLOUD_PROJECT")
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		return nil
+	}
+
+	return &mrpb.MonitoredResource{
+		Type: "gae_app",
+		Labels: map[string]string{
+			"project_id":  projectID,
+			"module_id":   os.Getenv("GAE_SERVICE"),
+			"version_id":  os.Getenv("GAE_VERSION"),
+			"instance_id": os.Getenv("GAE_INSTANCE"),
+			"runtime":     os.Getenv("GAE_RUNTIME"),
+			"zone":        zone,
+		},
+	}
+}
+
+func isCloudFunction() bool {
+	// Reserved envvars in older function runtimes, e.g. Node.js 8, Python 3.7 and Go 1.11.
+	_, name := os.LookupEnv("FUNCTION_NAME")
+	_, region := os.LookupEnv("FUNCTION_REGION")
+	_, entry := os.LookupEnv("ENTRY_POINT")
+
+	// Reserved envvars in newer function runtimes.
+	_, target := os.LookupEnv("FUNCTION_TARGET")
+	_, signature := os.LookupEnv("FUNCTION_SIGNATURE_TYPE")
+	_, service := os.LookupEnv("K_SERVICE")
+	return (name && region && entry) || (target && signature && service)
+}
+
+func detectCloudFunction() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		return nil
+	}
+	// Newer functions runtimes store name in K_SERVICE.
+	functionName, exists := os.LookupEnv("K_SERVICE")
+	if !exists {
+		functionName, _ = os.LookupEnv("FUNCTION_NAME")
+	}
+	return &mrpb.MonitoredResource{
+		Type: "cloud_function",
+		Labels: map[string]string{
+			"project_id":    projectID,
+			"region":        regionFromZone(zone),
+			"function_name": functionName,
+		},
+	}
+}
+
+func isCloudRun() bool {
+	_, config := os.LookupEnv("K_CONFIGURATION")
+	_, service := os.LookupEnv("K_SERVICE")
+	_, revision := os.LookupEnv("K_REVISION")
+	return config && service && revision
+}
+
+func detectCloudRunResource() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		return nil
+	}
+	return &mrpb.MonitoredResource{
+		Type: "cloud_run_revision",
+		Labels: map[string]string{
+			"project_id":         projectID,
+			"location":           regionFromZone(zone),
+			"service_name":       os.Getenv("K_SERVICE"),
+			"revision_name":      os.Getenv("K_REVISION"),
+			"configuration_name": os.Getenv("K_CONFIGURATION"),
+		},
+	}
+}
+
+func isKubernetesEngine() bool {
+	clusterName, err := metadata.InstanceAttributeValue("cluster-name")
+	// Note: InstanceAttributeValue can return "", nil
+	if err != nil || clusterName == "" {
+		return false
+	}
+	return true
+}
+
+func detectKubernetesResource() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	zone, err := metadata.Zone()
```

Comment:
You should check to make sure this is in "us-central1-a" format, rather than "projects/xxxx/regions/us-central1-a". When  you click the "Logs" button in the GKE UI, it will assume the first format.

I had this issue recently for Cloud Functions here: https://github.com/googleapis/python-logging/pull/291

Reply:
Good pt, I double checked this. Go library formats it for us already :P